lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -106,6 +124,26 @@ def initialize(driver_url, config)
         logger: Chef::Log.logger,
         retry_limit: Chef::Config.chef_provisioning[:aws_retry_limit] || 5
       )
+
+      driver = self
+      Chef::Resource::Machine.send(:define_method, :aws_object) do
```

Comment:
I would also like feedback on this - it adds the `aws_object` method to the base resources which will simplify life for _everyone_.  It means they will be applied even if the driver is switched, but we already made that caveat for the `aws_tags`.  Again, this is only a hack until we move away from the base resources.